CoorpAcademy/mobile
# segregate signin with QR Code from magic link

## Detailed purpose of the PR



PRODUCT : The need is to be able to distinguish a signing using the QR Code from a signin through the magic link.

TECH : just pass a Authorization type to the signing redux action and GTM will do the rest.

HOW TO TEST :

*QRCODE* :
- Grab a test phone
-  log on your firebase account
- go to the debug view
- open your favourite web platform and scan the qr code
- then you should see the signin event on firebase with the right `authenticationType`

*MAGIC-LINK* : 

- same as QRCODE but you will need to use the magic link authentication method :-)


### Result and observation

LOOK at the `AuthenticationType` => qr code :-)

<img width="613" alt="Screenshot 2019-05-16 at 18 27 14" src="https://user-images.githubusercontent.com/15958334/57870869-4ee1c080-7808-11e9-94c9-8c29f939bcae.png">

And here look we've signed in using the magic link 

<img width="600" alt="Screenshot 2019-05-16 at 18 31 22" src="https://user-images.githubusercontent.com/15958334/57871149-f8c14d00-7808-11e9-94f9-45f6a282608d.png">






       If checked, what have you broken ?

- [ ] **Extra dependency ?**
      If checked, Which extra dependency did you add ? (name, purpose, link ...).

### Testing Strategy

- [X] Already covered by tests
